server-extension/src/main/java/flyway/paikkis/V2_18__link_selkokartta_to_bglayer_selector.java

```diff
@@ -0,0 +1,108 @@
+        }
+        return null;
+    }
+
+    private void updateLayerExternalId(Connection conn, int layerId, String externalId) throws SQLException {
+        final String sql = "UPDATE oskari_maplayer SET externalid=? WHERE id=?";
+        try (PreparedStatement statement = conn.prepareStatement(sql)) {
+            statement.setString(1, externalId);
+            statement.setInt(2, layerId);
+            statement.execute();
+        }
+    }
+
+    private void updateViewsWithBaselayerSelector(Connection conn, String externalId) throws SQLException {
+        final String sql = "SELECT view_id, seqno, config FROM portti_view_bundle_seq WHERE bundleinstance = 'mapfull'";
+        try (PreparedStatement statement = conn.prepareStatement(sql)) {
+            try (ResultSet rs = statement.executeQuery()) {
+                while (rs.next()) {
+                    JSONObject config = JSONHelper.createJSONObject(rs.getString("config"));
+                    updateOneViewIfNeeded(conn, rs.getInt("view_id"), rs.getInt("seqno"), config, externalId);
+                }
+            }
+        }
+    }
+
+    private void updateOneViewIfNeeded(Connection conn, int viewId, int seqNo, JSONObject config, String externalId) throws SQLException {
+        JSONArray plugins = JSONHelper.getJSONArray(config, "plugins");
+        for (int i = 0; i < plugins.length(); i++) {
+            try {
+                JSONObject plugin = (JSONObject) plugins.get(i);
+                if (PLUGIN_ID.equals(JSONHelper.getStringFromJSON(plugin, "id", null))) {
+                    JSONObject pluginConfig = JSONHelper.getJSONObject(plugin, "config");
+                    if (pluginConfig == null) {
+                        continue;
+                    }
+                    JSONArray baseLayers = JSONHelper.getJSONArray(pluginConfig, "baseLayers");
+                    if (baseLayers == null) {
+                        continue;
+                    }
+
+                    boolean hasSelkoLayer = false;
+                    for (int j = 0; j < baseLayers.length(); j++) {
+                        if (externalId.equals(baseLayers.getString(j))) {
+                            hasSelkoLayer = true;
+                        }
+                    }
+                    if (hasSelkoLayer) {
+                        continue;
+                    }
+                    baseLayers.put(externalId);
+
+                    final String sql = "UPDATE portti_view_bundle_seq SET config=? WHERE view_id=? AND seqno=?";
+                    try (PreparedStatement statement = conn.prepareStatement(sql)) {
+                        statement.setString(1, config.toString(2));
+                        statement.setInt(2, viewId);
+                        statement.setInt(3, seqNo);
+                        statement.execute();
+                    }
+                }
+            } catch (JSONException e) {
+                LOG.error("Error in migration: {}!", e.getMessage());
```

Comment:
{} isn't replaced with e.getMessage(). You can just do:

               LOG.error("Error in migration:", e.getMessage());